lib/ecto/adapters/postgres.ex

```diff
@@ -343,7 +343,7 @@ defmodule Ecto.Adapters.Postgres do
     args =
       if port = opts[:port], do: ["-p", to_string(port)|args], else: args
 
-    host = opts[:hostname] || System.get_env("PGHOST") || "localhost"
+    host = opts[:socket] || opts[:socket_dir] || opts[:hostname] || System.get_env("PGHOST") || "localhost"
```

Comment:
as mentioned in the issue, we can't use a :socket as host so let's either ignore, raise, or warn on it.

```suggestion
    host = opts[:socket_dir] || opts[:hostname] || System.get_env("PGHOST") || "localhost"
```